source/MoorDyn2.hpp

```diff
@@ -432,13 +465,17 @@ class MoorDyn : public io::IO
 
 	/// The time integration scheme
 	TimeScheme* _t_integrator;
+	/// The time integration scheme name
```

Comment:
The time integrators have an inline method to retrieve the name, so this is quite redundant. If you need this while parsing the input file, I think it should be rather a local variable

Reply:
Agreed. I think the reason for this originally being implemented this way is that in the current setup, timeschemes are created using a create_time_scheme() function in the MoorDyn2::readInFile() member function as opposed to the MoorDyn2::readOptionsLine() member function, so we can just change the time integrator to be setup there and only do the default setup with RK2 if that options line isn't specified.

More generally, is there a reason to keep the time integrator setup in the create_time_scheme() function as opposed to the constructor (or maybe a builder)?

Reply:
Well, `create_time_scheme()` is almost a map which gets a name and returns a time scheme. However, there are some special cases, like the implicit schemes. That's why I made it a function. On the other hand I think it is quite appropriate to hold it on Time.hpp and Time.cpp.

Anyway, if you find it a better place or implementation in general, I obviously do not oppose